x-pack/plugins/ml/public/application/model_management/delete_models_modal.tsx

```diff
@@ -102,54 +111,90 @@ export const DeleteModelsModal: FC<DeleteModelsModalProps> = ({ models, onClose
+                    />
+                  }
+                  checked={deletePipelines}
+                  onChange={setDeletePipelines.bind(null, (prev) => !prev)}
+                  data-test-subj="mlModelsDeleteModalDeletePipelinesCheckbox"
+                />
+              </div>
+              <ul>
+                {modelsWithPipelines.flatMap((model) => {
+                  return Object.keys(model.pipelines).map((pipelineId) => (
+                    <li key={pipelineId}>{pipelineId}</li>
+                  ));
+                })}
+              </ul>
+            </EuiCallOut>
+            <EuiSpacer size="m" />
+          </>
+        ) : null}
+
+        {modelsWithInferenceAPIs.length > 0 ? (
           <EuiCallOut
             title={
               <FormattedMessage
-                id="xpack.ml.trainedModels.modelsList.deleteModal.pipelinesWarningHeader"
-                defaultMessage="{modelsCount, plural, one {{modelId} has} other {# models have}} associated pipelines."
+                id="xpack.ml.trainedModels.modelsList.deleteModal.inferenceAPIWarningHeader"
+                defaultMessage="{modelsCount, plural, one {{modelId} has} other {# models have}} associated inference services."
                 values={{
-                  modelsCount: modelsWithPipelines.length,
-                  modelId: modelsWithPipelines[0].model_id,
+                  modelsCount: modelsWithInferenceAPIs.length,
+                  modelId: modelsWithInferenceAPIs[0].model_id,
                 }}
               />
             }
             color="warning"
             iconType="warning"
           >
+            <ul>
+              {inferenceAPIsIDs.map((inferenceAPIModelId) => (
+                <li key={inferenceAPIModelId}>{inferenceAPIModelId}</li>
+              ))}
+            </ul>
+
             <div>
               <p>
                 <FormattedMessage
-                  id="xpack.ml.trainedModels.modelsList.deleteModal.warningMessage"
-                  defaultMessage="Deleting the trained model and its associated {pipelinesCount, plural, one {pipeline} other {pipelines}} will permanently remove these resources. Any process configured to send data to the {pipelinesCount, plural, one {pipeline} other {pipelines}} will no longer be able to do so once you delete the {pipelinesCount, plural, one {pipeline} other {pipelines}}. Deleting only the trained model will cause failures in the {pipelinesCount, plural, one {pipeline} other {pipelines}} that {pipelinesCount, plural, one {depends} other {depend}} on the model."
-                  values={{ pipelinesCount }}
+                  id="xpack.ml.trainedModels.modelsList.deleteModal.warningInferenceMessage"
+                  defaultMessage="Deleting only the trained model will cause failures in the inference {inferenceAPIsCount, plural, one {service} other {services}} that {inferenceAPIsCount, plural, one {depends} other {depend}} on the model."
```

Comment:
Seeing as there isn't an option to delete the inference service (as there is with pipelines), I'd remove the word `only` from `Deleting only` here.

<img width="598" alt="image" src="https://github.com/elastic/kibana/assets/7405507/21d003ec-4187-44fe-9fab-61e22738ec94">

Reply:
Updated https://github.com/elastic/kibana/pull/175880/commits/4793d17b2f3c1da6e97736b5c9a1ba5164540ad9